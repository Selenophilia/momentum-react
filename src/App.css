.App {
  text-align: center;
}


.header h1{
  font-size: 64px;
  font-weight: 300;
  margin:  20px 0 25px;
  color: #2c3e50;
  text-align:  center;
}

.header h1 span{
  font-weight: 700;
}

.header h2{
  color: #bdc3c7;
  font-weight: 300;
  text-align:  center;
}