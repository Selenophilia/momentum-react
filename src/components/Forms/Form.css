.todo__input{
    width: 30em;
    height: 60px;
    background: none;
    border: none;
    border-bottom: 1px solid #34495e;
    outline: none;
    font-size: 28px;
    padding: 5px;
    margin-top: 1em;
    margin-left: 3em;
}