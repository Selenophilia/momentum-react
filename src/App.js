import React, { Component, useState } from 'react';
import './App.css';
import Forms from  './components/Forms/Form'
import Todo from './components/Todos/Todos'

function App() {
  const [todos, setTodos] = useState([
                            { id: 0, name: 'eat'},
                            { id: 1, name: 'sleep'},
                            { id: 2, name: 'repeat'}    
                          ])

    const addTodo = (name) => {
     let newTodos = [...todos];
     let id = todos.length + 1 - 1;
     newTodos.push({id, name})
      setTodos(newTodos)
    } 

   const deleteTodo = (id) => {
      let todosArr = [...todos];
      let getId = todos.findIndex( t  =>  t.id === id)
      
      todosArr.splice(getId, 1)
      setTodos(todosArr)
    }
   const editTodo = (id, newName) => {
      const newTodo = todos.map(todo => {
        if(todo.id === id){
          return{...todo, name: newName}
        }
        return{...todo}
       })
      setTodos({  todos:newTodo })
    }

  const getTodo = todos.map(( todo ) => 
                            <Todo key={todo.id}
                                  name={todo.name}
                                  id={todo.id}
                                  deleteTodo={deleteTodo}
                                  editTodo={editTodo}  
                                  />
                      ); 
               
  const displayText = (
     <h1 className='display-text'>Nothing To Do here!!</h1>
  );  
  

  return (
    <div>
       <div className="header">
            <h1>Todo<span>List</span></h1>
            <h2>waddup this is a simple todoList </h2>
        
        </div>
        <div className='sample'>
        <Forms 
                  addTodo={addTodo}
                  />
        { getTodo }
        </div>
    </div>
      );

}

export default App;
