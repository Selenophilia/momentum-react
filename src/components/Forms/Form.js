import React, { Component } from 'react'
import './Form.css'

class Form extends Component{
    state = {
        todoInput: ''
    };
    handleInput = (e) => {
        this.setState({
            todoInput: e.target.value
        })
    }

    handleSubmit = (e) => {
        e.preventDefault();
        this.props.addTodo(this.state.todoInput)
        
        this.setState({
            todoInput: ""
        })

    }
    render(){
        return(
            <div>
                <form onSubmit={this.handleSubmit}>
                    <input 
                          className='todo__input'  
                          type='text'
                          name='text'  
                          placeholder='Insert your task here...'
                          value={this.state.todoInput}        
                          onChange={this.handleInput}
                    />

                </form>
             </div>
        );
    }

}


export default  Form;
