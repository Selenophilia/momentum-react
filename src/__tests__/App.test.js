import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../App';
import Form from   '../components/Forms/Form'


it('renders input form for todos ', () => {
  const { container } = render(<Form />)
  container.querySelector('.todo__input')
  expect(container.firstChild).toMatchSnapshot()
});

it('renders all todos ', () => {
  const { container } = render(<App />)
  container.querySelector('.todos')
  expect(container.firstChild).toMatchSnapshot()
});

