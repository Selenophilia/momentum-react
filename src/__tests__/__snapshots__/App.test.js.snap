// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all todos  1`] = `
<div>
  <div
    class="header"
  >
    <h1>
      Todo
      <span>
        List
      </span>
    </h1>
    <h2>
      waddup this is a simple todoList 
    </h2>
  </div>
  <div
    class="sample"
  >
    <div>
      <form>
        <input
          class="todo__input"
          name="text"
          placeholder="Insert your task here..."
          type="text"
          value=""
        />
      </form>
    </div>
    <div
      class="todos"
    >
      <ul
        class="todo__list"
      >
        <li>
          eat
          <span
            class="todo__delete"
          >
             X
          </span>
          <span
            class="todo__edit"
          >
             edit
          </span>
        </li>
      </ul>
    </div>
    <div
      class="todos"
    >
      <ul
        class="todo__list"
      >
        <li>
          sleep
          <span
            class="todo__delete"
          >
             X
          </span>
          <span
            class="todo__edit"
          >
             edit
          </span>
        </li>
      </ul>
    </div>
    <div
      class="todos"
    >
      <ul
        class="todo__list"
      >
        <li>
          repeat
          <span
            class="todo__delete"
          >
             X
          </span>
          <span
            class="todo__edit"
          >
             edit
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`renders input form for todos  1`] = `
<div>
  <form>
    <input
      class="todo__input"
      name="text"
      placeholder="Insert your task here..."
      type="text"
      value=""
    />
  </form>
</div>
`;
